src/services/errorRecoveryManager.ts

```
@@ -0,0 +1,564 @@
+    maxBackoffMs: 10000,
+    circuitBreakerThreshold: 5,
+    circuitBreakerTimeoutMs: 30000,
+    enableAutoRecovery: true,
+    logAllErrors: true
+  };
+
+  private readonly maxHistorySize = 100; // Keep memory usage low
+
+  constructor() {
+    if (ErrorRecoveryManager.instance) {
+      return ErrorRecoveryManager.instance;
```

Biome (2.1.2)

[error] The constructor should not return a value.

The constructor is here:

Returning a value from a constructor may confuse users of the class.

(lint/correctness/noConstructorReturn)